### DISCUSSION TITLE
DA Capstone | SyntaxError: f-string: unmatched '['

### DISCUSSION THREAD

**Discussion by goomba-roomba** [teamalgoritma/community#662] [Q&A] [answered]
I'm not really sure what to fix, I've tried wrapping `f` different combinations of `""` and `''` and `"""` and `'''` but nothing is working.

Thanks in advance.

<img width="1050" alt="Screenshot 2023-05-15 at 01 34 14" src="https://github.com/teamalgoritma/community/assets/129946460/c9a2d4e5-4b59-4416-84aa-ead86127fa7e">

**Comment by rusdipermana** [comment, marked as answer]
HI, you can try this

![may](https://github.com/teamalgoritma/community/assets/6158006/29304fd5-f5d4-4749-bb73-f0cce673abce)

The syntax error in this line of code is due to the fact that you are using single quotes to delimit the string that contains an f-string expression that also uses single quotes. This causes Python to interpret the f-string expression prematurely, resulting in an unterminated expression error.

To fix this issue, you can either use double quotes to delimit the outer string, or you can escape the inner quotes with a backslash character. Here is an example that should work:

`card_data = f"{df['Rate'].mean().round(2)}"`
